portal/src/main/webapp/js/src/data_download.js

```diff
@@ -176,10 +176,12 @@ var DataDownloadTab = (function() {
                     return a.localeCompare(b);
                 });
                 for (var j=0; j<sample_ids.length; j++) {
-                    strs.case_matrix += studyIds[i] + "\t";
-                    strs.case_matrix += sample_ids[j] + "\t";
-                    strs.case_matrix += (altered_study_sample_map[studyIds[i]][sample_ids[j]] ? "1" : "0") + "\t";
-                    strs.case_matrix += "\n";
+                    if (_.keys(altered_study_sample_map[studyIds[i]]) > 0) {
```

Comment:
it would be better to make a variable to hold `altered_study_sample_map[studyIds[i]]` so we dont keep accessing it, and pull it out of the loop since it doesnt change as `j` changes.